File: app/services/query_engine.py
"""Backward-compatible facade for query parse + dispatch.

Prefer importing from ``query_parse`` (pure) or ``query_dispatch`` (execute).
"""
from app.services.query_dispatch import (
    QueryEngine,
    SearchContext,
    SearchResult,
    execute_search,
    search_words,
)
from app.services.query_parse import (
    CodeTailQuery,
    CompoundAntQuery,
    DigitCodeQuery,
    EqualsQuery,
    HybridCodeQuery,
    HybridTailEqualsAliasQuery,
    JyutpingFragmentQuery,
    LiteralRefQuery,
    MaskQuery,
    ParsedQuery,
    QueryKind,
    RelationLookupQuery,
    RhymeAnchorQuery,
    UnmatchedQuery,
    WordLookupQuery,
    parse_query,
)

__all__ = [
    "CodeTailQuery",
    "CompoundAntQuery",
    "DigitCodeQuery",
    "EqualsQuery",
    "HybridCodeQuery",
    "HybridTailEqualsAliasQuery",
    "JyutpingFragmentQuery",
    "LiteralRefQuery",
    "MaskQuery",
    "ParsedQuery",
    "QueryEngine",
    "QueryKind",
    "RelationLookupQuery",
    "RhymeAnchorQuery",
    "SearchContext",
    "SearchResult",
    "UnmatchedQuery",
    "WordLookupQuery",
    "execute_search",
    "parse_query",
    "search_words",
]

File: client/src/db/query-engine.ts
/**
 * Query engine facade — re-exports client/src/db/query/* (Arch Phase B).
 */
export { QueryKind, RouteKind } from './query-kind.ts';
export {
  normalizeQuery,
  normalizeAndParse,
  parseQuery,
  parseHeteronymCodeQuery,
  parseDoubledSyllableSyntax,
  parseRelationSyntax,
  parseCodeRefMiddleRhymeQuery,
  parseTripleRhymeAnchorQuery,
  parseWildcardCodeAnchorQuery,
  tryParseBeforeMask,
  isRelationSyntaxQuery,
  parseAtTailQuery,
  parsePlusAnchorQuery,
  parsePrefixWildcardEqualsQuery,
  parsePrefixWildcardInitialQuery,
  parsePartialRhymeMaskQuery,
  parsePartialInitialMaskQuery,
  parseSerialPhonemeAnchorQuery,
  parseJyutpingAnchorQuery,
  parseRhymeAnchorQuery,
  isFramedEqualsQuery,
  isPingZeSerialQuery,
  normalizeQuerySyntax,
  JYUTPING_SYN_MODE_HINT,
  CODE_TAIL_MIDDLE,
  parserLogicSelfCheck,
  lookupLayoutSelfCheck,
  codePrefixedWholeWordEqualsEmptyHint,
  resolveFallback0243Mode,
} from './query/parse.ts';
export { dispatchParsed, executeListFilter } from './query/dispatch.ts';
export { dispatchSynMode } from './query/mode-dispatch.ts';
export { buildLookupLayout } from './query/lookup-layout.ts';
export { QueryEngine, queryEngine, searchWords, executeSearch } from './query/engine.ts';

export type {
  QueryMode,
  ParsedQuery,
  QueryResult,
  SearchContext,
  SearchResult,
  DigitCodeQuery,
  HeteronymCodeQuery,
  JyutpingAnchorQuery,
  WordLookupQuery,
  JyutpingFragmentQuery,
  MaskQuery,
  RelationLookupQuery,
  UnmatchedQuery,
} from './query-types.ts';

export { executeCanonicalMatchSpecPage } from './position-match/engine.ts';
export {
  canonicalMatchSpecToJson,
  finalizeCanonicalMatchSpec,
} from './position-match/canonical.ts';
export { compileParsedQuery, compileQuery, requireMatchSpecQuery } from './position-match/compiler.ts';
export {
  getCandidatesForLength,
  LengthCodeCandidateSource,
  positionMatchSourcesSelfCheck,
} from './position-match/sources.ts';
export type { CanonicalMatchSpec } from './position-match/canonical.ts';
